src/main/java/duke/parser/DukeParser.java

```diff
@@ -0,0 +1,235 @@
+package duke.parser;
+
+import java.time.format.DateTimeParseException;
+
+import duke.commands.*;
+import duke.storage.Storage;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import duke.exceptions.DukeException;
+import duke.tasklist.TaskList;
+
+
+
+/**
+ * Handler class that manages user input to duke.Duke.
+ * TODO more JavaDocs
+ */
+public class DukeParser {
+
+    private final String BREAK_LINES = "=================================";
+
+    private TaskList taskList;
+    private String keyword;
+    private String restOfInputString;
+
+    /**
+     * Default constructor for the DukeParser object.
+     *
+     * @param taskList A reference of duke.Duke's ArrayList of Tasks
+     */
+    public DukeParser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+
+    /**
+     * Handles user input, and preps parser for instruction execution.
+     *
+     * @param inputString The string that we would like to parse.
+     */
+    public void parseInput(String inputString)  {
+
+        // Sanitise user input first before continuing
+        String newString = sanitiseUserInput(inputString);
+
+        // Grab keyword from instruction
+        this.keyword = newString.split(" ")[0].toLowerCase();
+
+        // Split rest of input string for further parsing
+        // If there is no rest of input string, then restOfInputString will be "".
+        this.restOfInputString = newString.substring(this.keyword.length()).trim();
+
+    }
+
+    /**
+     * Executes a loaded and parsed instruction.
+     *
+     * @param st Storage that we want to save data to after the instruction has been executed
```

Comment:
Perhaps this param was changed? Now, should this param tag be needed?

Reply:
Yep, I should remove this param.